pkg/controller/kfservice/resources/credentials/service_account_credentials.go

```diff
@@ -114,13 +115,20 @@ func (c *CredentialBuilder) CreateSecretVolumeAndEnv(namespace string, serviceAc
 					Value: gcs.GCSCredentialVolumeMountPath + gcsCredentialFileName,
 				})
 		} else if secret.Name == azureSecretName {
-			log.Info("Setting secret envs for azure", "AzureSecret", secret.Name)
-			envs := azure.BuildSecretEnvs(secret, &c.config.AZURE)
-			container.Env = append(container.Env, envs...)
+			// prefer the named secret to any other secret
```

Comment:
Seems still keep the secret default name way, not sure if need. Shall we just keep 2nd way that's consistent with gcs and s3? Since for the case: user env has a secret named `azcreds`, but user want to use another secret and attached with sa, and specified the sa for KFServing, if so, KFServing always use `azcreds`, and never get the new secret which is specifed by sa unless removed `azcreds` but may be used by others.